library/alloc/src/string.rs

```diff
@@ -2944,19 +3185,20 @@ impl Iterator for Drain<'_> {
 }
 
 #[stable(feature = "drain", since = "1.6.0")]
-impl DoubleEndedIterator for Drain<'_> {
+impl<A: Allocator> DoubleEndedIterator for Drain<'_, A> {
     #[inline]
     fn next_back(&mut self) -> Option<char> {
         self.iter.next_back()
     }
 }
 
 #[stable(feature = "fused", since = "1.26.0")]
-impl FusedIterator for Drain<'_> {}
+impl<A: Allocator> FusedIterator for Drain<'_, A> {}
 
 #[cfg(not(no_global_oom_handling))]
 #[stable(feature = "from_char_for_string", since = "1.46.0")]
 impl From<char> for String {
+    // FIXME(zachs18): A: Allocator + Default?
```

Comment:
I don't think we should add this comment; there's many APIs which *could* do this (even on Vec) and it doesn't seem worth leaving comments strewn around std for that future possibility.

Reply:
Yeah, I had intended to remove or implement most of the `FIXME`s based on feedback before the PR gets merged. I'll just remove the ones that don't currently exist for `Vec` for this PR.